### ISSUE TITLE
extension is not working with SSO

### ISSUE THREAD

**Issue by cycomanic** [linkwarden/browser-extension#45]
The extension is not working with SSO (oauth2) at the moment.  No rush (I know oauth2 just landed), just wanted to document it here.

**Comment by crkos** [comment]
Hello, I just pushed into dev a version that fixes this, If you can install it in your browser and let me know if it's working for you that would be great!

**Comment by crkos** [comment]
Make sure to check the new options on the Config window on the extension

**Comment by apo2k7** [comment]
Hi! Thank you for adding the SSO login feature. I tried it with the Brave browser. Worked perfectly!

**Comment by laxmanpradhan** [comment]
any chance this version could be pushed to the firefox add-on store? Thanks!

https://addons.mozilla.org/en-US/firefox/addon/linkwarden/

**Comment by crkos** [comment]
Hello, I'm currently working on an OpenAPI specification of the API for linkwarden, I haven't tested this for firefox which will be definitly not work, because i havent modified the manifest and i'm not sure some browser apis i'm using work with firefox and V2 extensions, until i havent done that, i can't upload to the store, However it will be done soon. Worst case i have to mantain two versions of this.

**Comment by alceasan** [comment]
I could help with testing if there is any available build to install.

**Comment by crkos** [comment]
If you have npm you can compile it
with `npm run build` then do as the readme says, just install the folder name dist

**Comment by chrisbsmith** [comment]
This worked for me (thanks!), but I got a warning that I needed to supply a username/password even with the `Use SSO` checkbox checked.

**Comment by crkos** [comment]
Yeah thats actually something i was thinking about, since it will shift the content every time you would click on it, I just leave it there, but if you prefer like that i can just render it based on that, I actually doesnt matter if you provide username/password when SSO is checked is just UI validation nothing is sent to the server othen than checking if there is a session active

**Comment by crkos** [comment]
Currently i dont have much time to work on it, Whats left to jump to next version is working the firefox (manifest v2...) version, it shouldn't take much effort is just modifying the manifest and a few things, It will be soon on the stores to just add it to browser, also a few features are already in, such as omnibox, context right click to add the page and automatic sync even if you dont add from the extension but the local bookmark it will be added to linkwarden

**Comment by chrisbsmith** [comment]
> Yeah thats actually something i was thinking about, since it will shift the content every time you would click on it (or just make them not required), I just leave it there, but if you prefer like that i can just render it based on that, I actually doesnt matter if you provide username/password when SSO is checked is just UI validation nothing is sent to the server othen than checking if there is a session active.

I agree with your thoughts, but when I tried to auth with SSO checked, the validation failed that the username/password boxes were not populated.

**Comment by crkos** [comment]
Yes that is intented to happen, Even though not needed

**Comment by crkos** [comment]
I will probably remove it in the release

**Comment by JesseWebDotCom** [comment]
Is there a solution or workaround to this issue?

**Comment by CallMeTerdFerguson** [comment]
I'm guessing I updated to the version patched with SSO at some point and forgot, so if so apologies, but did either the last official version or the version pushed last week contain SSO? I got an update pushed via the web store (chrome) and the new version doesn't seem to have SSO options. Do I need to rebuild the old patched version and downgrade? Is SSO coming to mainline anytime soon?

**Comment by daniel31x13** [comment]
This issue will get fixed soon.

**Comment by EHRETic** [comment]
> Version 1.2.8 of the extension is now out and fixes this issue. It's already published on Firefox, on Chrome it needs to be approved (takes less than a day). You can use an access token to sign into the extension.

Hi there 😉

So if I understand correctly, we have to create a token in order to sign-in into the extension (I have the 1.2.8 installed today)?

If this is, I'm sorry to "complain", but that is disassociating the two component/sessions. 
Meaning if I sign-off from Linkwarden, my extention is still active until token is expired. 

It is not an huge security risk but it is still one that we didn't have before 😊
